src/charm.py

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env python3
+# Copyright 2023 Mariyan Dimitrov
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+"""Charm the service.
+
+Refer to the following post for a quick-start guide that will help you
+develop a new k8s charm using the Operator Framework:
+
+https://discourse.charmhub.io/t/4208
+"""
+
+import collections
+import logging
+import os
+
+import ops.lib
+from charms.nginx_ingress_integrator.v0.nginx_route import require_nginx_route
+from ops.charm import CharmBase
+from ops.framework import StoredState
+from ops.main import main
+from ops.model import ActiveStatus, WaitingStatus
+
+# Log messages can be retrieved using juju debug-log
+logger = logging.getLogger(__name__)
+
+pgsql = ops.lib.use("pgsql", 1, "postgresql-charmers@lists.launchpad.net")
+DATABASE_NAME = "synapse"
+
+VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
+
+
+class MatrixOperatorCharm(CharmBase):
+    """Charm the service."""
+
+    _CONTAINER_NAME = "synapse"
+    _SYNAPSE_CONFIG_PATH = "/data/homeserver.yaml"
+
+    _stored = StoredState()
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.start, self._on_install)
+        self.framework.observe(
+                self.on.synapse_pebble_ready,
+                self._on_config_changed
+        )
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.register_user_action, self._register_user)
+        require_nginx_route(
+            charm=self,
+            service_hostname=self._external_hostname,
+            service_name=self.app.name,
+            service_port=8008,
+        )
+        self._stored.set_default(
+            db_name=None,
+            db_user=None,
+            db_password=None,
+            db_host=None,
+            db_port=None,
+        )
+
+        self.db = pgsql.PostgreSQLClient(self, "db")  # pylint: disable=C0103
```

Comment:
We have guidance on how to disable linting issues, is it applicable here: https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#static-code-analysis?

Reply:
Also applies for any other linting rule disables

Reply:
Thanks, that's from reusing code from one of our charms, I intend to address these when having CI in place, I'm currently not running the code through a linter.